### PR TITLE
MDG Public API python wrapper

### PR DESCRIPTION
**Pull request by william-lukusa**
Story details : [[ch58204]](https://app.clubhouse.io/dataiku/story/58204)

⚠️ use and merge with https://github.com/dataiku/dip/pull/12168

### CODE REVIEW

File: dataikuapi/dss/ml.py [hunk 1]
```diff
@@ -1752,6 +1752,84 @@ def get_diagnostics(self):
...
+        """
+        Start the export of the model documentation from a docx template (as a file object).
+
+        :param object fp: A file-like object pointing to a template docx file
+        :return: A :class:`~dataikuapi.dss.future.DSSFuture` representing the model document generation process
+        """
+        files = {'file': fp}
+        if self.mltask is not None:
+            f = self.mltask.client._perform_json(
+                "POST", "/projects/%s/models/lab/%s/%s/models/%s/model-document/export-documentation-from-custom-template" %
```
Comment (FlorentDev64) [inline dataikuapi/dss/ml.py:1792]: /model-document/export-documentation-from-custom-template seems not a valid URL

**Review by FlorentDev64 — CHANGES_REQUESTED** [review]

File: dataikuapi/dss/ml.py (hunk 1 above)
Comment (william-lukusa) [reply dataikuapi/dss/ml.py:1792]: Ah indeed 👍 changed the url but missed the extra `model-document/`

**Review by FlorentDev64 — APPROVED** [review]

**Review by FlorentDev64 — APPROVED** [review]
LGTM

**Review by FlorentDev64 — COMMENTED** [review]
I think that this two routes should be merged into one single route to avoid code duplication.
`"GET", "/projects/%s/models/lab/documentations/%s" % (self.mltask.project_key, export_id))`
`"GET", "/projects/%s/savedmodels/documentations/%s" % (self.saved_model.project_key, export_id))`

Because, we do not need to know where the documentation came from (`savedModels` or 'models/labs`).
Maybe we should have one single route like 
`"GET", "/projects/%sdocumentations/%s" % (self.saved_model.project_key, export_id))`

If this is done, we should move the route to another file.

**Comment by william-lukusa** [comment]
> Because, we do not need to know where the documentation came from (`savedModels` or 'models/labs`). Maybe we should have one single route like `"GET", "/projects/%sdocumentations/%s" % (self.saved_model.project_key, export_id))`

@pbailly @adescamps what do you think ? We discussed it and I still think having the route `/documentations/{exportId}`  in projects might be confusing and keeping it like that is better in the documentation. But I'd like to have some more feedback 😄

**Pull request merged**